cpp/core/jni/JniWrapper.cc

```diff
@@ -566,6 +567,38 @@ Java_io_glutenproject_vectorized_NativeColumnarToRowJniWrapper_nativeClose(JNIEn
   JNI_METHOD_END()
 }
 
+JNIEXPORT void JNICALL
+Java_io_glutenproject_vectorized_NativeRowToColumnarJniWrapper_nativeConvertRowToColumnar(
+    JNIEnv* env,
+    jobject,
+    jlong cSchema,
+    jlongArray row_length,
+    jlong memory_address,
+    jlong allocator_id,
+    jlong cArray) {
+  JNI_METHOD_START
+  if (row_length == nullptr) {
+    gluten::JniThrow("Native convert row to columnar: buf_addrs can't be null");
+  }
+  int num_rows = env->GetArrayLength(row_length);
+  jlong* in_row_length = env->GetLongArrayElements(row_length, JNI_FALSE);
+  uint8_t* address = reinterpret_cast<uint8_t*>(memory_address);
+  auto* allocator = reinterpret_cast<MemoryAllocator*>(allocator_id);
+  if (allocator == nullptr) {
+    gluten::JniThrow("Memory pool does not exist or has been closed");
+  }
+  auto pool = AsWrappedArrowMemoryPool(allocator);
+  std::shared_ptr<arrow::Schema> schema =
+      gluten::JniGetOrThrow(arrow::ImportSchema(reinterpret_cast<struct ArrowSchema*>(cSchema)));
+
+  std::shared_ptr<arrow::RecordBatch> rb;
+  auto converter = std::make_shared<gluten::RowToColumnarConverter>(schema, num_rows, in_row_length, address, pool.get());
+  JniAssertOkOrThrow(converter->Init(&rb), "Native convert Row to Columnar Init RowToColumnarConverter failed");
```

Comment:
`converter->Init(&rb)` looks like some sort of initialization of the converter? When does the converting process actually happen?

Reply:
Or just some inaccurate naming?